rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/common/CommonRest.java

```diff
@@ -0,0 +1,187 @@
+            return false;
+        }
+    }
+
+    @Override
+    public String currentUser(String sessionId) {
+        try {
+            return getUserName(sessionId);
+        } catch (NotConnectedRestException e) {
+            return null;
+        }
+    }
+
+    @Override
+    public UserData currentUserData(String sessionId) {
+        return scheduler().getUserDataFromSessionId(sessionId);
+    }
+
+    @Override
+    public boolean portalAccess(String sessionId, String portal) throws NotConnectedRestException {
+        Scheduler scheduler = checkAccess(sessionId, "EMPTY");
+
+        try {
+            return checkPermission(scheduler.getSubject(),
+                                   new PortalAccessPermission(portal),
+                                   "Acess to portal " + portal + " is disabled");
+        } catch (PermissionException e) {
+            return false;
+        } catch (NotConnectedException e) {
+            throw new NotConnectedRestException("you are not connected to the scheduler, you should log on first");
+        }
+    }
+
+    /**
+     * the method check if the session id is valid i.e. a scheduler client is
+     * associated to the session id in the session map. If not, a
+     * NotConnectedRestException is thrown specifying the invalid access *
+     *
+     * @return the scheduler linked to the session id, an
+     * NotConnectedRestException, if no such mapping exists.
+     * @throws NotConnectedRestException
+     */
+    private Scheduler checkAccess(String sessionId, String path) throws NotConnectedRestException {
+        Session session = sessionStore.get(sessionId);
+
+        Scheduler schedulerProxy = session.getScheduler();
+
+        if (schedulerProxy == null) {
+            throw new NotConnectedRestException(YOU_ARE_NOT_CONNECTED_TO_THE_SCHEDULER_YOU_SHOULD_LOG_ON_FIRST);
+        }
+
+        renewSession(sessionId);
+
+        return schedulerProxy;
+    }
+
+    /**
+     * Call a method on the scheduler's frontend in order to renew the lease the
+     * user has on this frontend. see PORTAL-70
+     *
+     * @throws NotConnectedRestException
+     */
+    protected void renewSession(String sessionId) throws NotConnectedRestException {
+        try {
+            SharedSessionStore.getInstance().renewSession(sessionId);
+        } catch (NotConnectedException e) {
+            throw new NotConnectedRestException(YOU_ARE_NOT_CONNECTED_TO_THE_SCHEDULER_YOU_SHOULD_LOG_ON_FIRST, e);
+        }
+    }
+
+    /**
+     * Checks if user has the specified permission.
+     *
+     * @return true if it has, throw {@link SecurityException} otherwise with specified error message
+     */
+    private boolean checkPermission(final Subject subject, final Permission permission, String errorMessage)
+            throws PermissionException {
+        try {
+            Subject.doAsPrivileged(subject, new PrivilegedAction<Object>() {
+                public Object run() {
```

Comment:
I am sure you can replace this with Lambda.

Reply:
ok

Reply:
Thanks